src/components/Dialog/index.stories.tsx

```diff
@@ -153,6 +156,72 @@ export const WithLongBody = () => (
 
 WithLongBody.storyName = 'With long body'
 
+const Search = styled.div`
+  padding: ${space[16]};
+  border-block-end: 1px solid ${color.stroke};
+
+  @media ${device.mobileL} {
+    padding-block: ${space[32]};
+    padding-inline: ${space[32]};
+  }
+`
+
+const Flex = styled.div`
+  display: flex;
+  flex-direction: column;
+  grid-gap: ${space[8]};
+
+  @media ${device.tablet} {
+    grid-gap: ${space[16]};
+  }
+`
+
+export const WithScroll = ({
+  children,
+  ...props
+}: {
+  children: ReactNode
+  persist?: boolean
+  defaultOn?: boolean
+}) => (
+  <Dialog {...props}>
+    {({ hide, getToggleProps, getWindowProps }) => (
+      <>
+        <Button {...getToggleProps()}>Show dialog</Button>
+        <DialogWindow {...getWindowProps()}>
+          <DialogHeader>
+            Dialog Title
+            {!props.persist && (
+              <DialogAdornment right>
+                <button onClick={hide}>
+                  <CloseAlt size={24} />
+                </button>
+              </DialogAdornment>
+            )}
+          </DialogHeader>
+          <Search>
+            <Input
+              id="preferredCurrency"
+              startAdornment={<MagnifyingGlass size={24} />}
+              aria-label="A search field"
+              placeholder="Search for your currency"
+            />
+          </Search>
```

Comment:
If we have a header, body and footer I don't think it makes sense to have content that is not in one of those components. I would rather add this to the header, and then you probably need to adjust the header component a bit.

Reply:
Good one, yeah I agree this feels like I've taken an existing structure and messed it up a bit. Gonna look into it :)